### PR TITLE
MM-T661  /invite extra white space before @ in DM or GM

### PR DESCRIPTION
**Pull request by clarmso**
#### Summary

Extra spaces between `/invite` and `@` should not prevent the users from inviting other users to a channel.

#### Ticket Link

https://automation-test-cases.vercel.app/test/MM-T661

#### Related Pull Requests

- Has server changes N/A
- Has redux changes N/A
- Has mobile changes N/A

#### Screenshots

Screen Shot 2020-08-28 at 12 25 30 AM

### CODE REVIEW

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 1]
```diff
@@ -288,6 +288,35 @@ describe('I18456 Built-in slash commands: common', () => {
         cy.uiWaitUntilMessagePostedIncludes(`You were added to the channel by @${user1.username}`);
     });
 
+    it('MM-T661 /invite extra white space before @ in DM or GM', () => {
+        const user = userGroup[6];
+        const userToInviteGM = userGroup[5];
+        const userToInviteDM = userGroup[4];
+
+        cy.apiAddUserToChannel(testChannel.id, user.id);
+        loginAndVisitDefaultChannel(user, `${team1.name}/channels/${testChannel.name}`);
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:297]: I think it should be a GM not a regular channel.

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 2]
```diff
@@ -288,6 +288,35 @@ describe('I18456 Built-in slash commands: common', () => {
...
+        cy.get('#postListContent', {timeout: TIMEOUTS.HALF_MIN}).should('be.visible');
+
+        // # In a GM use the /invite command to invite a user to a channel you have permission to dd them to but place extra white space before the username
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:300]: I've corrected the typo in test case.
```suggestion
        // # In a GM use the /invite command to invite a user to a channel you have permission to add them to but place extra white space before the username
```

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 3]
```diff
@@ -288,6 +288,35 @@ describe('I18456 Built-in slash commands: common', () => {
...
+        cy.uiWaitUntilMessagePostedIncludes(`@${userToInviteGM.username} added to the channel by you`);
+
+        cy.get('#addDirectChannel').click();
+        cy.get('#selectItems').type(`${userToInviteDM.username}`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.get('#multiSelectList').findByText(`@${userToInviteDM.username}`).click();
+        cy.findByText('Go').click();
+        cy.get('#channelHeaderDropdownButton').contains(`${userToInviteDM.username}`).should('be.visible');
+
+        // # In a DM use the /invite command to invite a user to a channel you have permission to dd them to but place extra white space before the username
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:313]:
```suggestion
        // # In a DM use the /invite command to invite a user to a channel you have permission to add them to but place extra white space before the username
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js (hunk 3 above)
Comment (clarmso) [reply e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js]: Whoops! Fixed now.

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js (hunk 2 above)
Comment (clarmso) [reply e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js]: Whoops! Fixed now.

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 4]
```diff
@@ -288,6 +288,36 @@ describe('I18456 Built-in slash commands: common', () => {
...
+        // # In a GM use the /invite command to invite a user to a channel you have permission to add them to but place extra white space before the username
+        cy.postMessage(`/groupmsg @${userGroup[0].username} @${userGroup[1].username}`);
+        cy.postMessage(`/invite        @${userToInviteGM.username} ~${testChannel.name}`);
+
+        // * User added to channel as expected
+        cy.uiWaitUntilMessagePostedIncludes(`${userToInviteGM.username} added to ${testChannel.name} channel.`);
+
+        cy.get('#addDirectChannel').click();
+        cy.get('#selectItems').type(`${userToInviteDM.username}`);
+        cy.wait(TIMEOUTS.ONE_SEC);
```
Comment (clarmso) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:309]: Could someone give me a better suggestion than waiting 1 second?  If I don't insert this wait, cypress tells me the DOM has been detached.  Visually the result of the search appears, disappears and reappears.

Comment (saturninoabril) [reply e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:309]: One way is to do other assertion, like:
```javascript
cy.findByText('Loading').should('be.visible');
cy.findByText('Loading').should('not.exist');
```

**Review by saturninoabril — APPROVED** [review]
Thanks @clarmso, LGTM!

**Pull request merged**